OpenKeychain/src/main/java/org/sufficientlysecure/keychain/receiver/NetworkReceiver.java

```diff
@@ -0,0 +1,54 @@
+package org.sufficientlysecure.keychain.receiver;
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import org.sufficientlysecure.keychain.service.KeyserverSyncAdapterService;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import org.sufficientlysecure.keychain.Constants;
+import org.sufficientlysecure.keychain.util.Log;
+
+public class NetworkReceiver extends BroadcastReceiver {
+    private static final String ACTION_UPDATE_ALL = "update_all";
```

Comment:
The purpose of NetworkReceiver is to initiate sync as soon as wifi is restored. UPDATE_ALL will ensure that happens while SYNC_NOW will delay the sync till the screen is turned off. Since the sync has already been delayed (due to absence of wifi) I did not want to delay it any further. Also, the user might turn off the wifi before turning the screen off which would cause further delay in sync.